counter/src/main/java/org/infinispan/counter/impl/CounterModuleLifecycle.java

```diff
@@ -102,6 +103,7 @@ private static void registerCounterManager(GlobalComponentRegistry registry, Com
             counterManager = new EmbeddedCounterManager(future,
                   registry.getGlobalConfiguration().globalState().enabled());
             registry.registerComponent(counterManager, CounterManager.class);
+            registry.getComponent(CacheManagerJmxRegistration.class).start(); //HACK!
```

Comment:
Looks like a hack indeed, but why is it needed?

Reply:
added the following comment
```
            //this start() is only invoked when the DefaultCacheManager.start() is invoked
            //it is invoked here again to force it to check the managed global components 
            // and register them in the MBeanServer, if they are missing.
```
let me know if it is clear or not